### PR TITLE
[ENGPROD-6377] Supersede BRANCHING.yaml with RELEASE.yaml for federalized release

### PR DESCRIPTION
**Pull request by julenpardo**
## Summary

As part of the [PBMP-5643 - Federalizing the Release Process initiative](https://datarobot.atlassian.net/browse/PBMP-5643), we require each repository to define a RELEASE.yaml file.

So we're replacing "todo checklist" for humans to manually follow with a more formal definition that allows to perform a branch cut in an automated and repeatable.

**References**

[Distributed Release Metadata - RELEASE.yaml for 9.2 technical conversation](https://datarobot.atlassian.net/wiki/spaces/Ignite/pages/5618761738/Distributed+Release+Metadata+-+RELEASE.yaml)
[RELEASE.yaml specification](https://dev-docs.drdev.io/release/branches/RELEASE.yaml.html)

### CODE REVIEW

File: RELEASE.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
+description: |
```
Comment (rvorobii) [inline RELEASE.yaml:1]: @julenpardo please note that this is a public repo -- please make sure it's ok to expose such information.

also pinging @yakov-g for review

Comment (julenpardo) [reply RELEASE.yaml:1]: Yes right, I double checked with @carsongee before

**Review by rvorobii — APPROVED** [review]

**Comment by julenpardo** [comment]
jenkins run black_check

**Comment by julenpardo** [comment]
jenkins run test_integration_general

**Comment by carsongee** [comment]
jenkins run test_integration_general

**Pull request merged**